native/src/main/java/io/ballerina/stdlib/http/api/BallerinaHTTPConnectorListener.java

```diff
@@ -139,7 +139,8 @@ private boolean executeInterceptorServices(List<HTTPInterceptorServicesRegistry>
 
     private boolean checkForInterceptorDataBinding(HttpCarbonMessage inboundMessage, int interceptorServiceIndex,
                                                    InterceptorResource interceptorResource) {
-        if (HttpDispatcher.shouldDiffer(interceptorResource) && inboundMessage.isAccessedInNonInterceptorService()) {
+        if (!inboundMessage.isLastHttpContentArrived() && HttpDispatcher.shouldDiffer(interceptorResource) &&
```

Comment:
Looks like the isLastHttpContentArrived is synchronised. Shall we look for any performance change doing 10 mins load test?

Reply:
I think it's not needed as the same thread executes this invocation

Reply:
Created a task for this : [`Perf test for payload binding in HTTP service #3079`](https://github.com/ballerina-platform/ballerina-standard-library/issues/3079)